src/main/java/org/jenkinsci/plugins/azurekeyvaultplugin/AzureCredentialsProvider.java

```diff
@@ -124,9 +125,11 @@ private static Collection<IdCredentials> fetchCredentials() {
                         tags = new HashMap<>();
                     }
 
-                    if (StringUtils.isNotBlank(labelSelector) && !labelSelector.equals(tags.get("jenkins-label"))) {
-                        // User specified a label selector in config, but current credential does not contain a matching tag, skip iteration
-                        continue;
+                    if (StringUtils.isNotBlank(labelSelector)) {
+                        String jenkinsLabels = tags.getOrDefault("jenkins-label", "");
+                        if (Arrays.asList(jenkinsLabels.split(",")).indexOf(labelSelector) == -1) {
```

Comment:
This would be easier to read I think:
```suggestion
                        List<String> labelSelectors = Arrays.asList(jenkinsLabels.split(","));
                        if (!labelSelectors.contains(labelSelector)) {
```

Reply:
Works for me, I'll update.

Reply:
Change made. Reverified the change against our existing Jenkins/Azure setup.